Sources/FirebaseRemoteCommand.swift

```diff
@@ -237,103 +185,156 @@ public class FirebaseRemoteCommand: RemoteCommand {
-            "param_value": AnalyticsParameterValue,
-            "param_virtual_currency_name": AnalyticsParameterVirtualCurrencyName,
-            "param_user_signup_method": AnalyticsUserPropertySignUpMethod,
-            "param_user_allow_ad_personalization_signals": AnalyticsUserPropertyAllowAdPersonalizationSignals
-        ]
-        return eventParameters[paramName] ?? paramName
+        return FirebaseRemoteCommand.eventParameters[paramName] ?? paramName
     }
 
 }
 
-fileprivate extension Dictionary where Key == String, Value == Any {
-    func extractItems() -> [[String: Any]]? {
-        [FirebaseItem](from: self)?.dictionaryArray
+extension Dictionary where Key == String, Value == Any {
+    
+    func normalizeParamsArrays() -> [String: [Any]] {
+        self.filter { $0.key.starts(with: "param_item") || $0.key == "param_quantity" || $0.key == "param_price" }
+            .reduce(into: [String: [Any]]()) { result, dictionary in
+                result[dictionary.key] = dictionary.value as? [Any] ?? [dictionary.value]
+            }
     }
-    func itemsToArray() -> [String: Any] {
-        self.reduce(into: [String: Any]()) { result, dictionary in
-            if dictionary.key.contains("item_") ||
-                dictionary.key.contains("quantity") ||
-                dictionary.key.contains("price") {
-                guard let _ = dictionary.value as? [Any] else {
-                    result[dictionary.key] = [dictionary.value]
-                    return
+    
+    func itemArraysToArrayOfItems() -> [[String: Any]] {
+        let itemParamsArrays = normalizeParamsArrays()
+        // We assume that all arrays have the same length
+        let count = itemParamsArrays.first?.value.count ?? 0
+        var result = Array<[String: Any]>(repeating: [:], count: count)
+        for i in 0 ..< count {
+            for key in itemParamsArrays.keys {
+                if let values = itemParamsArrays[key], values.count > i {
+                    result[i][key] = values[i]
                 }
             }
         }
+        return result
     }
 }
+
+extension FirebaseRemoteCommand {
+    static let eventParameters = [
+        "param_achievement_id": AnalyticsParameterAchievementID,
+        "param_ad_format": AnalyticsParameterAdFormat,
+        "param_ad_network_click_id": AnalyticsParameterAdNetworkClickID,
+        "param_ad_platform": AnalyticsParameterAdPlatform,
+        "param_ad_source": AnalyticsParameterAdSource,
+        "param_ad_unit_name": AnalyticsParameterAdUnitName,
+        "param_affiliation": AnalyticsParameterAffiliation,
+        "param_cp1": AnalyticsParameterCP1,
+        "param_campaign": AnalyticsParameterCampaign,
+        "param_campaign_id": AnalyticsParameterCampaignID, // Version 8.12.1
+        "param_character": AnalyticsParameterCharacter,
+//            "param_checkout_step": AnalyticsParameterCheckoutStep,
+//            "param_checkout_option": AnalyticsParameterCheckoutOption,
```

Comment:
Should we remove the commented ones? Or should they not be commented?

Reply:
should be removed as they are removed from the firebase APIs